app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialPresenter.java

```diff
@@ -0,0 +1,225 @@
+                    app.getVercode()))
+            .retry())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(created -> {
+        }, error -> {
+        });
+  }
+
+  private void resumeDownload() {
+    view.getLifecycle()
+        .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
+        .flatMap(create -> view.resumeDownload()
+            .flatMap(__ -> permissionManager.requestDownloadAccess(permissionService)
+                .flatMap(success -> permissionManager.requestExternalStoragePermission(
+                    permissionService))
+                .flatMapSingle(__1 -> editorialManager.loadEditorialViewModel())
+                .flatMapCompletable(
+                    app -> editorialManager.resumeDownload(app.getMd5(), app.getPackageName(),
+                        app.getAppId()))
+                .retry()))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(created -> {
+        }, error -> {
+        });
+  }
+
+  private void pauseDownload() {
+    view.getLifecycle()
+        .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
+        .flatMap(create -> view.pauseDownload()
+            .flatMapSingle(__ -> editorialManager.loadEditorialViewModel())
+            .doOnNext(app -> editorialAnalytics.sendDownloadPauseEvent(app.getPackageName()))
+            .flatMapCompletable(app -> editorialManager.pauseDownload(app.getMd5()))
+            .retry())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(created -> {
+        }, error -> {
+        });
+  }
+
+  private Completable downloadApp(DownloadModel.Action action, String packageName, long appId) {
+    return Observable.defer(() -> {
+      if (editorialManager.shouldShowRootInstallWarningPopup()) {
+        return view.showRootInstallWarningPopup()
+            .doOnNext(editorialManager::allowRootInstall)
+            .map(__ -> action);
+      }
+      return Observable.just(action);
+    })
+        .observeOn(viewScheduler)
+        .flatMap(__ -> permissionManager.requestDownloadAccess(permissionService))
+        .flatMap(success -> permissionManager.requestExternalStoragePermission(permissionService))
+        .observeOn(Schedulers.io())
+        .flatMapCompletable(__1 -> editorialManager.loadEditorialViewModel()
+            .flatMapCompletable(viewModel -> editorialManager.downloadApp(action, viewModel)))
+        .toCompletable();
+  }
+
+  private void loadDownloadApp() {
+    view.getLifecycle()
+        .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
+        .flatMap(created -> view.isAppViewReadyToDownload())
+        .flatMap(create -> editorialManager.loadEditorialViewModel()
+            .toObservable())
+        .flatMap(
+            app -> editorialManager.loadDownloadAppViewModel(app.getMd5(), app.getPackageName(),
+                app.getVercode(), false, null))
```

Comment:
Isn't it a bit dangerous to pass false and null? What if we have an editorial of paid apps?